bbc/simorgh
# WSTEAM1-240: Header image for WS Live

Resolves JIRA [WSTEAM1-240](https://jira.dev.bbc.co.uk/browse/WSTEAM1-240) - Header image for WS Live

Overall changes
======
Renders an Image in the header on WS Live pages when one is supplied. Otherwise renders a text-only header.

![Screenshot 2024-02-07 at 11 41 23](https://github.com/bbc/simorgh/assets/98817636/79974691-28a3-434e-baf9-7a48f59a1b52)

Code changes
======
- Extract required image values from the response and pass into the Header
- Create `isHeaderImage` boolean. This is true if the required image values are present
- Use this flag to render 'MaskedImage' component & apply required styles to the Header component
- Create MaskedImage component which renders Image with linear gradient image mask.
- Update Header styles to use Flex instead of Grid.
- Add tests and update snapshots
- Updates Image Component to not have a fixed Aspect Ratio if no width & height or aspect ratio is provided.
- Update other components affected by this and snapshots.
- Add extra long text storybook version for UX review.

Testing
======
Testing Locally
- pull down this branch
- cd into ws-nextjs-app and use yarn dev to run the application locally
- Visit http://localhost:7081/pidgin/live/c0mrezv7rd0t to see a header image (LTR)
- Visit http://localhost:7081/urdu/live/cd0jyl9r0lzt to see a header image (RTL)
- Visit http://localhost:7081/pidgin/live/c7p765ynk9qt to see a text-only header

Storybook links
- https://wsteam1-240-header-image--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=new-components-livepageheader--title-and-description-with-live-label-and-image&viewMode=story
- https://wsteam1-240-header-image--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&knob-Select%20a%20service=arabic&id=new-components-livepageheader--title-and-description-with-live-label-and-image&viewMode=story

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[UX designs (header image)](https://www.figma.com/file/D9w5JGLOhznDlz5tHu3rH0?node-id=5:63&locale=en&type=design)

[Screen reader UX Figma (header image)](https://www.figma.com/file/D9w5JGLOhznDlz5tHu3rH0/Live-mvp---Live-Topic-Header---handoff?type=design&node-id=104-10&mode=design&t=okUjISN0SwzmMfXw-0)

[Live Page Screen Reader UX](https://paper.dropbox.com/doc/WS-Live-page-Screen-reader-UX--CIFvUftYG2EdtFiNsaBzR9ovAg-8qm1UHDGVMhv5Qj9Q2mbi)

[Live Page A11y A/Cs](https://paper.dropbox.com/doc/Live-page-Accessibility-Acceptance-Criteria--CIHCd~3oDcJAv_Zt_ucLcbbIAg-KZQODmAf9tvodUlxsAVnA)

[UX designs (text only header)](https://www.figma.com/file/ozHsWG5R9tETw6lBOU740V/Live-mvp-header-(text-only)---handover?type=design&node-id=1-2&mode=design&t=xxJSN2MlwLiK80Ei-0)

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
